numba/tests/test_array_manipulation.py

```diff
@@ -584,6 +588,38 @@ def check(pyfunc):
     def test_transpose_array_npm(self):
         self.test_transpose_array(flags=no_pyobj_flags)
 
+    def test_swapaxes_array(self, flags=enable_pyobj_flags):
+        a = np.arange(24).reshape(4, 3, 2)
+
+        # swap first and last
+        axis1 = 0
+        axis2 = -1
+        pyfunc = numpy_swapaxes_array
+        cr = compile_isolated(pyfunc, (typeof(a), typeof(axis1), typeof(axis2)), flags=flags)
+        cfunc = cr.entry_point
+        expected = pyfunc(a, axis1, axis2)
+        got = cfunc(a, axis1, axis2)
+        np.testing.assert_equal(expected, got)
+
+        # swap first and middle
+        axis1 = -3
+        axis2 = 1
+        pyfunc = numpy_swapaxes_array
+        cr = compile_isolated(pyfunc, (typeof(a), typeof(axis1), typeof(axis2)), flags=flags)
+        cfunc = cr.entry_point
+        expected = pyfunc(a, axis1, axis2)
+        got = cfunc(a, axis1, axis2)
+        np.testing.assert_equal(expected, got)
+
+        # check non-int axis raise an error
+        axis1 = 0
+        axis2 = 1.5
+        pyfunc = numpy_swapaxes_array
+        cr = compile_isolated(pyfunc, (typeof(a), typeof(axis1), typeof(axis2)), flags=flags)
+        cfunc = cr.entry_point
+        with self.assertRaises(TypeError):
```

Comment:
Check out the pattern used in Numba for this, there's an example  few lines down in `test_squeeze_array_npm`. Numba does `with self.assertRaises(<exception>) as e:` and then later checks that there's the expected error message in `e`.